### ISSUE TITLE
Quarkus Asciidoc plugin

### ISSUE THREAD

**Issue by rmanibus** [quarkusio/quarkus#43328]
### Description

Interested in this extension, please +1 via the emoji/reaction feature of GitHub (top right).

### Repository name

quarkus-asciidoc

### Short description

serve documentation along with your application 

### Repository Homepage URL

https://quarkiverse.github.io/quarkiverse-docs/quarkus-asciidoc/dev/

### Repository Topics

- quarkus-extension
- documentation 
...

### Team Members

- rmanibus

### Additional context

Objective:
- process Asciidoc file to html
- expose html in non application root (q/docs)

**Comment by mkouba** [comment]
@rmanibus FYI there is also a proposal to support asciidoc in [Qute Web](https://docs.quarkiverse.io/quarkus-qute-web/dev/index.html): https://github.com/quarkiverse/quarkus-qute-web/issues/95

It's not the same but I suppose that the functionality is very similar.

CC @mcruzdev

**Comment by rmanibus** [comment]
so the question is, do we want this as part of Qute web, or as a standalone extension (or both ?)

**Comment by rmanibus** [comment]
this seems to deviate a little bit from the initial objective of the Qute web extension (it does not really involve Qute)

**Comment by mkouba** [comment]
> this seems to deviate a little bit from the initial objective of the Qute web extension (it does not really involve Qute)

From my POV it's a subset of the proposed Qute web extension feature where the whole template is automatically surrounded by `{#asciidoc}`... That said, it's definitely not the same as I already mentioned in my previous comment. But my point is that the problem could be probably solved with that extension :shrug:.

**Comment by mcruzdev** [comment]
@rmanibus do you mean to expose a documentation located in a `docs` folder by example... and serve it with Quarkus? 
If so, we are working on a similar extension https://github.com/quarkiverse/quarkus-roq, this [one](https://pages.quarkiverse.io/quarkus-roq/)  was generated with Roq.

**Comment by rmanibus** [comment]
@mcruzdev yes that's exactly what I am looking for, that's awesome I will take a look !

**Comment by rmanibus** [comment]
@mcruzdev I took some time to dig a little bit into the Roq extension and it looks awesome ! 
The only missing piece is that I would love to generate my doc using [asciidoctorj](https://docs.asciidoctor.org/asciidoctorj/latest/) 

Do you have any recommendation on how to do this ? 
Should I contribute a new module similar to roq-frontmatter, using asciidoctorj ? 
Or could we use the roq-generator for this ?

**Comment by maxandersen** [comment]
There are two usecases here - one is having an extension that takes care of good asciidoctorj support - the other is the qute/web/roq integration. 

Given that asciidoctorj is *massive* when it comes to dependency chain and startup time but does offer larger feature set - would it make sense to  have both a `quarkus-asciidoctorj` extension and a `quarkus-asciidoc-java` extension which could be wired up to roq independently?

**Comment by maxandersen** [comment]
note, both of those could be in the same repo - just different variants.

**Comment by rmanibus** [comment]
I was not aware of Asciidoc java. Is it another implementation or just a wrapper around the original binary?

**Comment by maxandersen** [comment]
It's a pure java version - it is not trying to replicate all the functionality of asciidoctor.
